compiler/runtime/OMRRelocationRecord.cpp

```diff
@@ -0,0 +1,399 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <stdint.h>
+#include "omrcfg.h"
+#include "codegen/CodeGenerator.hpp"
+#include "env/FrontEnd.hpp"
+#include "codegen/Relocation.hpp"
+#include "compile/ResolvedMethod.hpp"
+#include "control/Options.hpp"
+#include "control/Options_inlines.hpp"
+#include "infra/Assert.hpp"
+#include "env/jittypes.h"
+#include "env/VMAccessCriticalSection.hpp"
+#include "infra/SimpleRegex.hpp"
+#include "runtime/CodeCache.hpp"
+#include "runtime/CodeCacheManager.hpp"
+#include "runtime/TRRelocationRecord.hpp"
+#include "runtime/TRRelocationRuntime.hpp"
+#include "runtime/TRRelocationTarget.hpp"
+#include "runtime/OMRRelocationRecord.hpp"
+
+#if defined(__IBMCPP__) && !defined(AIXPPC) && !defined(LINUXPPC)
+#define ASM_CALL __cdecl
+#else
+#define ASM_CALL
+#endif
+#if defined(TR_HOST_S390) || defined(TR_HOST_X86) // gotten from RuntimeAssumptions.cpp, should common these up
+extern "C" void _patchVirtualGuard(uint8_t *locationAddr, uint8_t *destinationAddr, int32_t smpFlag);
+#else
+extern "C" void ASM_CALL _patchVirtualGuard(uint8_t*, uint8_t*, uint32_t);
+#endif
+
+uint8_t
+OMR::RelocationRecordBinaryTemplate::type(TR::RelocationTarget *reloTarget)
+   {
+   return reloTarget->loadUnsigned8b(&_type);
+   }
+
+void
+OMR::RelocationRecordBinaryTemplate::setSize(TR::RelocationTarget *reloTarget,uint16_t size)
+   {
+      return reloTarget->storeUnsigned16b(size, reinterpret_cast<uint8_t*>(&_size));
+   }
+void
+OMR::RelocationRecordBinaryTemplate::setType(TR::RelocationTarget *reloTarget, uint8_t type)
+   {
+   return reloTarget->storeUnsigned8b(type,&_type);
+   }
+
+void
+OMR::RelocationRecordBinaryTemplate::setFlags(TR::RelocationTarget *reloTarget,uint8_t flags)
```

Comment:
any reason there isn't a `flags(...)` API that returns the flags?

Reply:
Hi, thank you for the question. This is a piece of code that we are not really using, but I think it's nice to have for continuing working in future. Shall we leave it or it's better to submit a minimal version?

Reply:
same as above.

Reply:
Thank you for the answer, Shall I mark this question as resolved?